src/views/open-api-docs-view/utils/get-response-data.ts

```diff
@@ -35,27 +35,35 @@ export async function getResponseData(
 		// Grab all the response messages for the current response code
 		const codeResponses = responses[responseCode]
 
-		// Process the parameters etc. for each of the response types (contentType)
-		// for the current response code (responseCode)
+		// Skip instances where codeResponses is a $ref
+		if ('$ref' in codeResponses) {
+			continue
+		}
 
-		for (const contentType in codeResponses['content']) {
-			const definition = codeResponses['content'][contentType]
+		// Skip instances where codeResponse does not have content
+		if (!codeResponses.content) {
+			continue
+		}
 
+		// Process the parameters etc. for each of the response types (contentType)
+		// for the current response code (responseCode)
+		for (const [contentType, definition] of Object.entries(
+			codeResponses.content
+		)) {
 			/**
 			 * Skip the current object if:
 			 *   - The response object is not JSON (for now)
 			 *   - The response object definition is a reference
 			 *	 - The response object does not include a definition or schema
 			 *	 - The response schema is a reference
 			 **/
-			const skipEntry =
+			if (
```

Comment:
This change allows TypeScript to correctly discriminate the union for `definition.schema`.